process/sync/baseSync.go

```diff
@@ -687,6 +698,48 @@ func (boot *baseBootstrap) syncBlock() error {
 	)
 
 	boot.cleanNoncesSyncedWithErrorsBehindFinal()
+	boot.cleanProofsBehindFinal(header)
+
+	return nil
+}
+
+func (boot *baseBootstrap) handleEquivalentProof(
+	header data.HeaderHandler,
+	headerHash []byte,
+) error {
+	if !boot.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, header.GetEpoch()) {
+		return nil
+	}
+
+	prevHeader, err := boot.blockBootstrapper.getHeaderWithHashRequestingIfMissing(header.GetPrevHash())
+	if err != nil {
+		return err
+	}
+
+	if !boot.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, prevHeader.GetEpoch()) {
+		// no need to check proof for first block after activation
+		log.Info("handleEquivalentProof: no need to check equivalent proof for first activation block")
+		return nil
+	}
+
+	// process block only if there is a proof for it
+	hasProof := boot.proofs.HasProof(header.GetShardID(), headerHash)
+	if hasProof {
+		return nil
+	}
+
+	log.Trace("baseBootstrap.handleEquivalentProof: did not have proof for header, will try again", "headerHash", headerHash)
+
+	// TODO: evaluate adding a wait here, or request for next header if missing
```

Comment:
not sure this would work as expected, to be seen in tests

Reply:
the integration test is working in this case, will see with more testing if there are other edge cases here